Sources/SwiftUIKit/Views/Map.swift

```diff
@@ -0,0 +1,453 @@
+                          animate: Bool = true,
+                          edgePadding: UIEdgeInsets? = nil
+                          ) -> Self {
+    if let padding = edgePadding {
+      setVisibleMapRect(rect, edgePadding: padding, animated: animate)
+    } else {
+      setVisibleMapRect(rect, animated: animate)
+    }
+    
+    return self
+  }
+  
+  /// Changes coordinates and span.
+  @discardableResult
+  public func move(to region: MKCoordinateRegion, animate: Bool = true) -> Self {
+    initialCoordinates = region.center
+    setRegion(region, animated: animate)
+    
+    return self
+  }
+  
+  
+  /// Changes only coordinates.
+  @discardableResult
+  public func move(to coordinates: CLLocationCoordinate2D, animate: Bool = true) -> Self {
+    let _region = MKCoordinateRegion(center: coordinates, span: region.span)
+    initialCoordinates = coordinates
+    setRegion(_region, animated: animate)
+    
+    return self
+  }
+  
+  @discardableResult
+  public func center(_ center: CLLocationCoordinate2D, animated: Bool = true) -> Self {
+    setCenter(center, animated: animated)
+    
+    return self
+  }
+  
+  @discardableResult
+  public func showAnnotations(_ annotations: [MKAnnotation], animated: Bool = true) -> Self {
+    super.showAnnotations(annotations, animated: animated)
+    
+    return self
+  }
+  
+  @discardableResult
+  public func showAnnotations(_ annotations: MKAnnotation..., animated: Bool = true) -> Self {
+    super.showAnnotations(annotations, animated: animated)
+    
+    return self
+  }
+}
+
+// MARK: - Constraining the Map View
+
+@available(iOS 13.0, *)
+extension Map {
+  @discardableResult
+  public func cameraBoundary(_ boundary: MKMapView.CameraBoundary?, animated: Bool = true) -> Self {
+    setCameraBoundary(boundary, animated: animated)
+    
+    return self
+  }
+  
+  @discardableResult
+  public func setCameraZoomRange(_ cameraZoomRange: MKMapView.CameraZoomRange?, animated: Bool) -> Self {
+    super.setCameraZoomRange(cameraZoomRange, animated: animated)
+    
+    return self
+  }
+}
+
+// MARK: - Configuring the Map's Appearance
+extension Map {
+  @discardableResult
+  public func camera(_ camera: MKMapCamera, animated: Bool = true) -> Self {
+    setCamera(camera, animated: animated)
+    
+    return self
+  }
+  
+  @discardableResult
+  public func showBuildings(_ bool: Bool? = nil) -> Self {
+    showsBuildings = bool ?? !showsBuildings
+    
+    return self
+  }
+}
+
+@available(iOS 13.0, *)
+extension Map {
+  @discardableResult
+  public func showCompass(_ bool: Bool? = nil) -> Self {
+    showsCompass = bool ?? !showsCompass
+    
+    return self
+  }
+  
+  @discardableResult
+  public func showScale(_ bool: Bool? = nil) -> Self {
+    showsScale = bool ?? !showsScale
+    
+    return self
+  }
+  
+  @discardableResult
+  public func showTraffic(_ bool: Bool? = nil) -> Self {
+    showsTraffic = bool ?? !showsTraffic
+    
+    return self
+  }
```

Comment:
I don't know if the optional logic is needed or expected... I'd prefer we just remove it being optional.

Reply:
[Commit](https://github.com/0xLeif/SwiftUIKit/pull/132/commits/9aa25b80cabfc3c27ac917f05559954444bd7019)